### PR TITLE
DocData: Skip unexposed classes

### PR DESCRIPTION
**Pull request by akien-mga**
Properly expose classes that we actually want accessible.

Thanks @neikeq for the hint.

---

Skips the following:
```
Class 'BulletPhysicsDirectBodyState3D' is not exposed, skipping.
Class 'BulletPhysicsServer3D' is not exposed, skipping.
Class 'GDScriptNativeClass' is not exposed, skipping.
Class 'IP_Unix' is not exposed, skipping.
Class 'PhysicsDirectBodyState2DSW' is not exposed, skipping.
Class 'PhysicsServer2DSW' is not exposed, skipping.
```
\+ `MonoGCHandle` in theory but I can't confirm as the Mono glue fails building right now.

### CODE REVIEW

File: core/crypto/crypto.h [hunk 1]
```diff
@@ -84,8 +84,6 @@ class Crypto : public Reference {
 };
 
 class ResourceFormatLoaderCrypto : public ResourceFormatLoader {
-	GDCLASS(ResourceFormatLoaderCrypto, ResourceFormatLoader);
```
Comment (akien-mga) [inline core/crypto/crypto.h:87]: This is to be consistent with #29872.
It now seems to me that simply skipping them in DocData as done in this PR would have been sufficient, so it's not clear to me what difference `GDCLASS` makes in this case.

File: servers/register_server_types.cpp [hunk 2]
```diff
@@ -100,18 +101,18 @@ void register_server_types() {
...
 	ClassDB::register_class<CameraServer>();
 
+	ClassDB::register_virtual_class<RenderingDevice>();
+
 	ClassDB::register_virtual_class<XRInterface>();
 	ClassDB::register_class<XRPositionalTracker>();
 
-	ClassDB::add_compatibility_class("ARVRServer", "XRServer");
-	ClassDB::add_compatibility_class("ARVRInterface", "XRInterface");
-	ClassDB::add_compatibility_class("ARVRPositionalTracker", "XRPositionalTracker");
```
Comment (akien-mga) [inline servers/register_server_types.cpp:113]: Moved to `scene/register_scene_types.cpp` for convenience as I'm consolidated everything there, even `server/` classes.

File: scene/register_scene_types.cpp [hunk 3]
```diff
@@ -857,12 +867,6 @@ void register_scene_types() {
 	ClassDB::add_compatibility_class("VisualShaderNodeScalarOp", "VisualShaderNodeFloatOp");
 	ClassDB::add_compatibility_class("VisualShaderNodeScalarUniform", "VisualShaderNodeFloatUniform");
 	ClassDB::add_compatibility_class("World", "World3D");
-	ClassDB::add_compatibility_class("ProceduralSky", "Sky");
-	ClassDB::add_compatibility_class("PanoramaSky", "Sky");
-	ClassDB::add_compatibility_class("ARVRCamera", "XRCamera3D");
-	ClassDB::add_compatibility_class("ARVROrigin", "XROrigin3D");
-	ClassDB::add_compatibility_class("ARVRController", "XRController3D");
-	ClassDB::add_compatibility_class("ARVRAnchor", "XRAnchor3D");
```
Comment (akien-mga) [inline scene/register_scene_types.cpp:865]: I'm amazed as I often seem to be the only one to notice when things seem to be ordered alphabetically on purpose... My brain must be wired differently ;)

File: core/crypto/crypto.h (hunk 1 above)
Comment (neikeq) [reply core/crypto/crypto.h:87]: A class without `GDCLASS` cannot be exposed, but the right way to do that is to just not call `ClassDB::register_*`.
That being said if the class doesn't need it, removing `GDCLASS` can be beneficial to reduce executable bloat, but it must be done with care as some classes depend on it (for `_bind_methods` for example). Also not sure how it affects inheritance, e.g.: `Object` <- `B` <- `C` where `C` has `GDCLASS` but `B` doesn't.

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**